### ISSUE TITLE
some questions

### ISSUE THREAD

**Issue by DsuNAmi** [yfchen1994/Teacher-Fingerprinting#1]
1. where is the example.sh?
2. what's the model_stealing?
3. waht's the pytorch.mode_provider, Then, sure for pytorchcv.model_provider?
4. Is it a code contained all things?

**Comment by DsuNAmi** [comment]
Thanks for your reply

**Comment by zmx666** [comment]
I have some questions by running your code.
Firstly,where is "pretrained_model_inference" in your PretrainedModel.py?
Secondly,where are "model_stealing"and "KnockoffDataset" in your stealing function?

**Comment by hupptechnologies** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
VaishuDhanapalan, @cooperz25, @shardulshinde, @bingforbetter, @ideaspad, @xxvunborn, @Elgatofurioso, @a186r, @LiuYangICV, @MojtabaTajik, @zfarbp, @A-Diddy, @hpw5, @hhy5277, @gooichi, @Nqubehm, @AchiNIN, @longcpp, @hans-riess, @Dergarcon, @nathyca, @DKilkenny, @xshg, @jtrcarlos, @Bombonchik, @mattiasaine, @mrberti, @enrixp, @vilarsail, @LookTech, @shin4141, @abaybek, @ChandreshJain, @0xHriday, @basnetsoyuj, @littlestone, @HanYouyang, @rhybs, @immiz, @On2Space, @amitkumarusc, @Deadkorskiy, @yorickshan, @Sam-Wang-1024, @cagev, @Jasonchang6435, @jw0903, @Tianny, @daniele2010, @bingo-V

**Comment by dhanyacp** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!

**Comment by yfchen1994** [comment]
@DsuNAmi Hi, this repo contains a cleaned implementation of the teacher fingerprinting attack:

1. Please see `example.h` in the top-level folder. 
2. The `model_stealing` directory contained experimental code used to demonstrate that, once the teacher model is identified, the cost of a model-stealing attack can be reduced. The code is experimental and not intended as a polished demo. During the cleanup I forgot to remove the dependency on model_stealing; sorry for any confusion from earlier commits.
3. Pretrained models are loaded via pytorchcv.model_provider using weights from this project: https://github.com/osmr/imgclsmob (Also see https://pypi.org/project/pytorchcv/)

**Comment by yfchen1994** [comment]
@zmx666 Hi, the previous commit of `PretrainedModel.py` included an unnecessary dependency on `pretrained_model_inference,` which was part of an experimental version. I’ve cleaned up the code to remove it.

The model-stealing code was not intended to be part of this polished demo, so I’ve removed it accordingly.
